executor/src/witgen/processor.rs

```diff
@@ -1,21 +1,47 @@
-use ast::analyzed::Identity;
+use std::{collections::HashSet, marker::PhantomData};
+
+use ast::analyzed::{Identity, PolyID, PolynomialReference, SelectedExpressions};
 use number::FieldElement;
 
+use crate::witgen::Constraint;
+
 use super::{
+    affine_expression::AffineExpression,
     identity_processor::IdentityProcessor,
     rows::{Row, RowFactory, RowPair, RowUpdater, UnknownStrategy},
     sequence_iterator::{
         DefaultSequenceIterator, IdentityInSequence, ProcessingSequenceIterator, SequenceStep,
     },
-    EvalError, FixedData,
+    Constraints, EvalError, EvalValue, FixedData,
 };
 
+type Left<'a, T> = Vec<AffineExpression<&'a PolynomialReference, T>>;
+
+// Marker types
+pub struct WithCalldata;
+pub struct WithoutCalldata;
+
+/// Data needed to handle an outer query.
+pub struct Calldata<'a, T: FieldElement> {
```

Comment:
In the code comments, you always call it "outer query", so why not also call the type like that?

Reply:
Sure, renamed it!